### PR TITLE
Dep versions

### PR DESCRIPTION
**Pull request by jonathanwgoodwin**
@gchomatas Allows blazar to have versions associated with Dependencies.
Does _not_ store them yet / no modifications have been made to the mysql queries.  

This adds explicit support for having versions of dependencies in blazar.
This will allow us to correctly do global dependency builds. 
Currently (with our internal maven discovery) A commit to a branch of some project would trigger builds of things that depend on that `groupId:artifactId` on master. Which is not what we want.

### CODE REVIEW

File: BlazarBase/src/main/java/com/hubspot/blazar/base/DependencyItem.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.hubspot.blazar.base;
+
+
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+public class DependencyItem {
```
Comment (jhaber) [inline BlazarBase/src/main/java/com/hubspot/blazar/base/DependencyItem.java]: Maybe just `Dependency`? Other things that come to mind are `Package` or `Artifact`

**Comment by gchomatas** [comment]
@jonathanwgoodwin can you add some background on why we are doing that and what we are expecting to solve.

**Comment by jonathanwgoodwin** [comment]
@gchomatas paas-build-deploy#175 has more detail. I've linked this PR there.

File: BlazarData/src/main/resources/schema.sql [hunk 2]
```diff
@@ -188,3 +188,7 @@ CREATE TABLE branch_settings (
   "interProjectBuildOptIn" TINYINT(1),
   PRIMARY KEY ("branchId")
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+--changeset jgoodwin:8
+ALTER TABLE module_provides ADD COLUMN "version" VARCHAR(190) AFTER "name";
+ALTER TABLE module_depends ADD COLUMN "version" VARCHAR(190) AFTER "name";
```
Comment (solomonty) [inline BlazarData/src/main/resources/schema.sql]: `VARCHAR(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci`

File: BlazarData/src/main/resources/schema.sql [hunk 3]
```diff
@@ -188,3 +188,7 @@ CREATE TABLE branch_settings (
   "interProjectBuildOptIn" TINYINT(1),
   PRIMARY KEY ("branchId")
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
Comment (solomonty) [inline BlazarData/src/main/resources/schema.sql]: `ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8`

Comment (jhaber) [inline BlazarData/src/main/resources/schema.sql]: Any MySQL-specific settings need to done using an `ALTER` after-the-fact in a MySQL-specific changeset, [for example](https://github.com/HubSpot/Blazar/blob/37a138384f56928fc9653f1cf678336b60561169/BlazarData/src/main/resources/schema.sql#L96-L97)

File: BlazarData/src/main/resources/schema.sql (hunk 2 above)
Comment (jhaber) [inline BlazarData/src/main/resources/schema.sql]: Any MySQL-specific settings need to done using an `ALTER` after-the-fact in a MySQL-specific changeset, [for example](https://github.com/HubSpot/Blazar/blob/37a138384f56928fc9653f1cf678336b60561169/BlazarData/src/main/resources/schema.sql#L96-L97)

Comment (solomonty) [inline BlazarData/src/main/resources/schema.sql]:
```
ALTER TABLE branch_settings ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8;

ALTER TABLE instant_message_configs ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8;

ALTER TABLE inter_project_build_mappings ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8;

ALTER TABLE inter_project_builds ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8;

ALTER TABLE malformed_files ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8;
```

File: BlazarData/src/main/resources/schema.sql (hunk 3 above)
Comment (jonathanwgoodwin) [inline BlazarData/src/main/resources/schema.sql]: Yep; was just going through this with monty.

**Pull request merged**